### PR TITLE
React group project

### PR DESCRIPTION
**Pull request by banzaman**
Hi There!

@banzaman  and @muhammedb8826  have made the following changes to the repository:

    Set up the Kanban project. ✨
    Create a React App and install tools like Redux, React Router, and Axios. 🔧
    Create routes on App.js to redirect to all the components on the SPA. 🔀
    Create directories to work with React Components and Redux Slices and files. 📄
    Fetch API data from SpaceX using Redux. ⬇️
    Configure store.js and write functionality to dispatch actions. 📈
    Render all react components with dynamic elements and API data. 🎨
    Add tests for components. ✅

Thank you so much for taking the time to review our code! ⚡️🔥

### CODE REVIEW

File: src/redux/missions/missionsSlice.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        state.isLoading = true;
+      })
+      .addCase(getMissions.fulfilled, (state, action) => {
+        const newState = action.payload.map((mission) => ({ ...mission, reserved: false }));
+        return ({
+          ...state,
+          missions: newState,
+          isLoading: false,
+        });
+      })
```
Comment (binhussen) [inline src/redux/missions/missionsSlice.js:48]: - Kindly, you are storing the whole API response object which is not necessary, Please filter and store only used data to redux store to improve its performance 👌

File: src/redux/rockets/rocketsSlice.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        }
+        return { ...item, reserved: true };
+      });
+      state.rocketItems = newRocketItems;
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(getRocketItems.fulfilled, (state, action) => {
+        state.isLoading = false;
```
Comment (binhussen) [inline src/redux/rockets/rocketsSlice.js:39]: - Kindly, you are storing the whole API response object which is not necessary, Please filter and store only used data to redux store to improve its performance 👌

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @banzaman and @muhammedb8826 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- All linter checks passed ✔️
- Used react-redux & testing-library ✔️
- Used Git flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ghost — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Github workflow
✅ No linter error
✅ Good Design

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by muhammedb8826** [comment]
Thank you for your time

**Pull request merged**